Add notes about smart sensors being deprecated in 2.2.4 (#340)
astronomer/airflow-guides: guides/what-is-a-sensor.md

## Smart Sensors

> **Note:** Smart sensors have been deprecated as of Airflow 2.2.4. It is generally recommended to use deferrable operators instead.

[Smart sensors](https://airflow.apache.org/docs/apache-airflow/stable/concepts/smart-sensors.html) are a relatively new feature released with Airflow 2.0, where sensors are executed in batches using a centralized process. This eliminates a major drawback of classic sensors, which use one process per sensor and therefore can consume considerable resources at scale for longer running tasks. With smart sensors, sensors in your DAGs are registered to the metadata database, and then a separate DAG fetches those sensors and manages processing them. 

![Smart Sensors](https://assets2.astronomer.io/main/guides/sensors-101/smart_sensors_architecture.png)

> Note that smart sensors were considered a beta feature, and have been deprecated as of Airflow 2.2.4. They are somewhat complex to implement, and not very widely used to-date. They are also generally less preferable to deferrable operators (more on these in the section below), which are more flexible.

Reviewer: ```suggestion
Note that smart sensors have been deprecated since Airflow 2.2.4. They are somewhat complex to implement and not very widely used to-date. They are also generally less preferable to deferrable operators (more on these in the section below), which are more flexible.
```
